Enable unit test execution from `npm test`

hideya opened the pull request:
Made `npm test` build the framework and run unit-tests on an emulator.

@rmg would you please start running `npm test` as a part of PR checks?
It makes sure that the framework gets built and all the unit tests runs successfully, but it doesn't tar up the resulted framework (as this is a test target).  Would you check the behavior on the CI please?

This PR includes a fix for a regression in [LBContainer createContainerWithName:..] which broke a unit test.  I included the fix together since a successful execution of the unit tests is necessary for testing the `npm test` script.

rmg commented on LoopBackTests/bin/run-unittests:
```diff
@@ -0,0 +1,32 @@
+
+# build the target
+xcodebuild \
+    -project LoopBack.xcodeproj \
```
The only difference I can see between this `xcodebuild` invocation and the one that CI currently does is CI adds `-derivedDataPath $WORKSPACE/derived`. I'm not overly familiar with the build process, but one thing we could do to unify this script with our traditional build process would be to add an `$XCODEBUILD_ARGS` in here so that the extra parameter could be injected.

hideya replied:
@rmg thank you for the review and the suggestion. Sounds good. I've added `$XCODEBUILD_ARGS` to the `xcodebuild` command line.  Does this look good?

bajtos commented:
Hi @hideya, thank you for this pull request!

I don't understand the low-level details enough to judge the implementation. I don't see any glaring issue and the test script passes on my machine, I assume it's all ok then.

However, I have few higher-level comments:

1. ATM, you copied dependencies from `LoopBackTests/server/package.json` to top-level package.json. Could you please remove `LoopBackTests/server/package.json` to prevent duplication? I think it's no longer needed.

2. How about `SLRemotingTests`? Are they being executed too?

3. There is also a folder `LoopBackOSXTests` containing `Info.plist` only. I don't remember details, is this just including existing LoopBack tests and running them on OSX? If it is so, then I'd like to include this in CI too.

hideya commented:
Hi @bajtos thanks for the careful review always!

As to 1., I've removed the obsolete `package.json` files.  Thanks for pointing it out.

As to 2., `SLRemotingTests` are executed as a part of `LoopBackTests` (and they can also be executed by themselves for testing the `SLRemoting` part only).  While checking the settings, I found some misconfiguration (`SLRemotingTests` target executing `LoopBackTests` too), and have fixed it.

As to 3., I've updated the test script to run the unit tests on OS X too (also fixed header file misconfigurations).

Do these sound good?

bajtos commented on package.json:
```diff
@@ -10,5 +10,20 @@
+    "url": "https://github.com/strongloop/loopback-sdk-ios/issues"
+  },
+  "scripts": {
+    "test": "LoopBackTests/bin/run-unittests"
```
nitpick: `run-unit-tests` would look better to my eyes.

bajtos commented:
Hi @hideya, thank you for the update, the patch LGTM.

Please consider cleaning up the history, merging most (if not all) of the CI-infrastructure related commits into a single one and fixing commit messages so that the first line is at most 50 characters long.

Feel free to land this whenever you consider it ready.

@rmg AFAICT from the [build console log](http://ci.strongloop.com/job/loopback-sdk-ios/158/console), the CI is not running these new tests yet. What is needed to make that happen?

hideya commented:
@bajtos I've cleaned up the commit history and am going to merge this PR.

> @rmg AFAICT from the build console log, the CI is not running these new tests yet. What is needed to make that happen?

@rmg please let me know if I need to do anything on my side.

rmg commented:
@slnode test please

rmg commented:
@slnode test please

rmg commented:
@slnode test please

rmg commented:
@bajtos @hideya good news is the tests are now being run... bad news is the build is failing.

Looks like a whole lot of this:
```
CompileC /Users/jenkins/Library/Developer/Xcode/DerivedData/LoopBack-eyexgcikeiibushkkvkmuctjtisk/Build/Intermediates/LoopBack.build/Debug-iphonesimulator/LoopBackTests.build/Objects-normal/x86_64/SLRESTAdapterTests.o SLRemotingTests/SLRESTAdapterTests.m normal x86_64 objective-c com.apple.compilers.llvm.clang.1_0.compiler
    cd /Users/jenkins/build/workspace/loopback-sdk-ios
    export LANG=en_US.US-ASCII
    export PATH="/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/usr/bin:/Applications/Xcode.app/Contents/Developer/usr/bin:/usr/local/opt/ruby/bin:/usr/local/bin:/usr/local/sbin:/usr/bin:/bin:/usr/sbin:/sbin"
    /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang -x objective-c -arch x86_64 -fmessage-length=0 -fdiagnostics-show-note-include-stack -fmacro-backtrace-limit=0 -std=gnu99 -fobjc-arc -Wno-trigraphs -fpascal-strings -O0 -Wno-missing-field-initializers -Wno-missing-prototypes -Wno-implicit-atomic-properties -Wno-receiver-is-weak -Wno-arc-repeated-use-of-weak -Wduplicate-method-match -Wno-missing-braces -Wparentheses -Wswitch -Wno-unused-function -Wno-unused-label -Wno-unused-parameter -Wunused-variable -Wunused-value -Wempty-body -Wuninitialized -Wno-unknown-pragmas -Wno-shadow -Wno-four-char-constants -Wno-conversion -Wconstant-conversion -Wint-conversion -Wno-bool-conversion -Wenum-conversion -Wshorten-64-to-32 -Wpointer-sign -Wno-newline-eof -Wno-selector -Wno-strict-selector-match -Wno-undeclared-selector -Wno-deprecated-implementations -DDEBUG=1 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator7.1.sdk -fexceptions -fasm-blocks -fstrict-aliasing -Wprotocol -Wdeprecated-declarations -g -Wno-sign-conversion -fobjc-abi-version=2 -fobjc-legacy-dispatch -mios-simulator-version-min=6.1 -iquote /Users/jenkins/Library/Developer/Xcode/DerivedData/LoopBack-eyexgcikeiibushkkvkmuctjtisk/Build/Intermediates/LoopBack.build/Debug-iphonesimulator/LoopBackTests.build/LoopBackTests-generated-files.hmap -I/Users/jenkins/Library/Developer/Xcode/DerivedData/LoopBack-eyexgcikeiibushkkvkmuctjtisk/Build/Intermediates/LoopBack.build/Debug-iphonesimulator/LoopBackTests.build/LoopBackTests-own-target-headers.hmap -I/Users/jenkins/Library/Developer/Xcode/DerivedData/LoopBack-eyexgcikeiibushkkvkmuctjtisk/Build/Intermediates/LoopBack.build/Debug-iphonesimulator/LoopBackTests.build/LoopBackTests-all-target-headers.hmap -iquote /Users/jenkins/Library/Developer/Xcode/DerivedData/LoopBack-eyexgcikeiibushkkvkmuctjtisk/Build/Intermediates/LoopBack.build/Debug-iphonesimulator/LoopBackTests.build/LoopBackTests-project-headers.hmap -I/Users/jenkins/Library/Developer/Xcode/DerivedData/LoopBack-eyexgcikeiibushkkvkmuctjtisk/Build/Products/Debug-iphonesimulator/include -I/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include -I/Users/jenkins/Library/Developer/Xcode/DerivedData/LoopBack-eyexgcikeiibushkkvkmuctjtisk/Build/Intermediates/LoopBack.build/Debug-iphonesimulator/LoopBackTests.build/DerivedSources/x86_64 -I/Users/jenkins/Library/Developer/Xcode/DerivedData/LoopBack-eyexgcikeiibushkkvkmuctjtisk/Build/Intermediates/LoopBack.build/Debug-iphonesimulator/LoopBackTests.build/DerivedSources -F/Users/jenkins/Library/Developer/Xcode/DerivedData/LoopBack-eyexgcikeiibushkkvkmuctjtisk/Build/Products/Debug-iphonesimulator -F/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator7.1.sdk/Developer/Library/Frameworks -F/Applications/Xcode.app/Contents/Developer/Library/Frameworks -F/Users/jenkins/build/workspace/loopback-sdk-ios -include /Users/jenkins/Library/Developer/Xcode/DerivedData/LoopBack-eyexgcikeiibushkkvkmuctjtisk/Build/Intermediates/PrecompiledHeaders/LoopBack-Prefix-gdhqsqsotndgqgcgodvhfishzvvv/LoopBack-Prefix.pch -MMD -MT dependencies -MF /Users/jenkins/Library/Developer/Xcode/DerivedData/LoopBack-eyexgcikeiibushkkvkmuctjtisk/Build/Intermediates/LoopBack.build/Debug-iphonesimulator/LoopBackTests.build/Objects-normal/x86_64/SLRESTAdapterTests.d --serialize-diagnostics /Users/jenkins/Library/Developer/Xcode/DerivedData/LoopBack-eyexgcikeiibushkkvkmuctjtisk/Build/Intermediates/LoopBack.build/Debug-iphonesimulator/LoopBackTests.build/Objects-normal/x86_64/SLRESTAdapterTests.dia -c /Users/jenkins/build/workspace/loopback-sdk-ios/SLRemotingTests/SLRESTAdapterTests.m -o /Users/jenkins/Library/Developer/Xcode/DerivedData/LoopBack-eyexgcikeiibushkkvkmuctjtisk/Build/Intermediates/LoopBack.build/Debug-iphonesimulator/LoopBackTests.build/Objects-normal/x86_64/SLRESTAdapterTests.o
/Users/jenkins/build/workspace/loopback-sdk-ios/SLRemotingTests/SLRESTAdapterTests.m:37:5: error: use of undeclared identifier 'XCTestExpectation'
    ASYNC_TEST_START
    ^
In file included from /Users/jenkins/build/workspace/loopback-sdk-ios/SLRemotingTests/SLRESTAdapterTests.m:13:
/Users/jenkins/build/workspace/loopback-sdk-ios/SLRemotingTests/../SLRemoting/SLRemotingTestsUtils.h:13:26: note: expanded from macro 'ASYNC_TEST_START'
#define ASYNC_TEST_START XCTestExpectation *expectation = [self expectationWithDescription:[NSString stringWithUTF8String:__FUNCTION__]];
                         ^
/Users/jenkins/build/workspace/loopback-sdk-ios/SLRemotingTests/SLRESTAdapterTests.m:37:5: error: use of undeclared identifier 'expectation'
In file included from /Users/jenkins/build/workspace/loopback-sdk-ios/SLRemotingTests/SLRESTAdapterTests.m:13:
/Users/jenkins/build/workspace/loopback-sdk-ios/SLRemotingTests/../SLRemoting/SLRemotingTestsUtils.h:13:45: note: expanded from macro 'ASYNC_TEST_START'
#define ASYNC_TEST_START XCTestExpectation *expectation = [self expectationWithDescription:[NSString stringWithUTF8String:__FUNCTION__]];
                                            ^
/Users/jenkins/build/workspace/loopback-sdk-ios/SLRemotingTests/SLRESTAdapterTests.m:37:5: error: no visible @interface for 'SLRESTAdapterTests' declares the selector 'expectationWithDescription:'
    ASYNC_TEST_START
    ^~~~~~~~~~~~~~~~
In file included from /Users/jenkins/build/workspace/loopback-sdk-ios/SLRemotingTests/SLRESTAdapterTests.m:13:
/Users/jenkins/build/workspace/loopback-sdk-ios/SLRemotingTests/../SLRemoting/SLRemotingTestsUtils.h:13:65: note: expanded from macro 'ASYNC_TEST_START'
#define ASYNC_TEST_START XCTestExpectation *expectation = [self expectationWithDescription:[NSString stringWithUTF8String:__FUNCTION__]];
                                                           ~~~~ ^~~~~~~~~~~~~~~~~~~~~~~~~~
/Users/jenkins/build/workspace/loopback-sdk-ios/SLRemotingTests/SLRESTAdapterTests.m:45:29: error: use of undeclared identifier 'expectation'
                            ASYNC_TEST_SIGNAL
                            ^
In file included from /Users/jenkins/build/workspace/loopback-sdk-ios/SLRemotingTests/SLRESTAdapterTests.m:13:
/Users/jenkins/build/workspace/loopback-sdk-ios/SLRemotingTests/../SLRemoting/SLRemotingTestsUtils.h:23:28: note: expanded from macro 'ASYNC_TEST_SIGNAL'
#define ASYNC_TEST_SIGNAL [expectation fulfill];
                           ^
/Users/jenkins/build/workspace/loopback-sdk-ios/SLRemotingTests/SLRESTAdapterTests.m:47:33: error: use of undeclared identifier 'expectation'
                        failure:ASYNC_TEST_FAILURE_BLOCK];
                                ^
In file included from /Users/jenkins/build/workspace/loopback-sdk-ios/SLRemotingTests/SLRESTAdapterTests.m:13:
/Users/jenkins/build/workspace/loopback-sdk-ios/SLRemotingTests/../SLRemoting/SLRemotingTestsUtils.h:31:6: note: expanded from macro 'ASYNC_TEST_FAILURE_BLOCK'
    [expectation fulfill]; \
     ^
/Users/jenkins/build/workspace/loopback-sdk-ios/SLRemotingTests/SLRESTAdapterTests.m:48:5: error: no visible @interface for 'SLRESTAdapterTests' declares the selector 'waitForExpectationsWithTimeout:handler:'
    ASYNC_TEST_END
    ^~~~~~~~~~~~~~
In file included from /Users/jenkins/build/workspace/loopback-sdk-ios/SLRemotingTests/SLRESTAdapterTests.m:13:
/Users/jenkins/build/workspace/loopback-sdk-ios/SLRemotingTests/../SLRemoting/SLRemotingTestsUtils.h:18:30: note: expanded from macro 'ASYNC_TEST_END'
#define ASYNC_TEST_END [self waitForExpectationsWithTimeout:10 handler:nil];
                        ~~~~ ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Users/jenkins/build/workspace/loopback-sdk-ios/SLRemotingTests/SLRESTAdapterTests.m:52:5: error: use of undeclared identifier 'XCTestExpectation'
    ASYNC_TEST_START
    ^
In file included from /Users/jenkins/build/workspace/loopback-sdk-ios/SLRemotingTests/SLRESTAdapterTests.m:13:
/Users/jenkins/build/workspace/loopback-sdk-ios/SLRemotingTests/../SLRemoting/SLRemotingTestsUtils.h:13:26: note: expanded from macro 'ASYNC_TEST_START'
#define ASYNC_TEST_START XCTestExpectation *expectation = [self expectationWithDescription:[NSString stringWithUTF8String:__FUNCTION__]];
                         ^
/Users/jenkins/build/workspace/loopback-sdk-ios/SLRemotingTests/SLRESTAdapterTests.m:52:5: error: use of undeclared identifier 'expectation'
In file included from /Users/jenkins/build/workspace/loopback-sdk-ios/SLRemotingTests/SLRESTAdapterTests.m:13:
/Users/jenkins/build/workspace/loopback-sdk-ios/SLRemotingTests/../SLRemoting/SLRemotingTestsUtils.h:13:45: note: expanded from macro 'ASYNC_TEST_START'
#define ASYNC_TEST_START XCTestExpectation *expectation = [self expectationWithDescription:[NSString stringWithUTF8String:__FUNCTION__]];
                                            ^
/Users/jenkins/build/workspace/loopback-sdk-ios/SLRemotingTests/SLRESTAdapterTests.m:52:5: error: no visible @interface for 'SLRESTAdapterTests' declares the selector 'expectationWithDescription:'
    ASYNC_TEST_START
    ^~~~~~~~~~~~~~~~
In file included from /Users/jenkins/build/workspace/loopback-sdk-ios/SLRemotingTests/SLRESTAdapterTests.m:13:
/Users/jenkins/build/workspace/loopback-sdk-ios/SLRemotingTests/../SLRemoting/SLRemotingTestsUtils.h:13:65: note: expanded from macro 'ASYNC_TEST_START'
#define ASYNC_TEST_START XCTestExpectation *expectation = [self expectationWithDescription:[NSString stringWithUTF8String:__FUNCTION__]];
                                                           ~~~~ ^~~~~~~~~~~~~~~~~~~~~~~~~~
/Users/jenkins/build/workspace/loopback-sdk-ios/SLRemotingTests/SLRESTAdapterTests.m:60:29: error: use of undeclared identifier 'expectation'
                            ASYNC_TEST_SIGNAL
                            ^
In file included from /Users/jenkins/build/workspace/loopback-sdk-ios/SLRemotingTests/SLRESTAdapterTests.m:13:
/Users/jenkins/build/workspace/loopback-sdk-ios/SLRemotingTests/../SLRemoting/SLRemotingTestsUtils.h:23:28: note: expanded from macro 'ASYNC_TEST_SIGNAL'
#define ASYNC_TEST_SIGNAL [expectation fulfill];
                           ^
/Users/jenkins/build/workspace/loopback-sdk-ios/SLRemotingTests/SLRESTAdapterTests.m:62:33: error: use of undeclared identifier 'expectation'
                        failure:ASYNC_TEST_FAILURE_BLOCK];
                                ^
In file included from /Users/jenkins/build/workspace/loopback-sdk-ios/SLRemotingTests/SLRESTAdapterTests.m:13:
/Users/jenkins/build/workspace/loopback-sdk-ios/SLRemotingTests/../SLRemoting/SLRemotingTestsUtils.h:31:6: note: expanded from macro 'ASYNC_TEST_FAILURE_BLOCK'
    [expectation fulfill]; \
     ^
/Users/jenkins/build/workspace/loopback-sdk-ios/SLRemotingTests/SLRESTAdapterTests.m:63:5: error: no visible @interface for 'SLRESTAdapterTests' declares the selector 'waitForExpectationsWithTimeout:handler:'
    ASYNC_TEST_END
    ^~~~~~~~~~~~~~
In file included from /Users/jenkins/build/workspace/loopback-sdk-ios/SLRemotingTests/SLRESTAdapterTests.m:13:
/Users/jenkins/build/workspace/loopback-sdk-ios/SLRemotingTests/../SLRemoting/SLRemotingTestsUtils.h:18:30: note: expanded from macro 'ASYNC_TEST_END'
#define ASYNC_TEST_END [self waitForExpectationsWithTimeout:10 handler:nil];
                        ~~~~ ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Users/jenkins/build/workspace/loopback-sdk-ios/SLRemotingTests/SLRESTAdapterTests.m:67:5: error: use of undeclared identifier 'XCTestExpectation'
    ASYNC_TEST_START
    ^
In file included from /Users/jenkins/build/workspace/loopback-sdk-ios/SLRemotingTests/SLRESTAdapterTests.m:13:
/Users/jenkins/build/workspace/loopback-sdk-ios/SLRemotingTests/../SLRemoting/SLRemotingTestsUtils.h:13:26: note: expanded from macro 'ASYNC_TEST_START'
#define ASYNC_TEST_START XCTestExpectation *expectation = [self expectationWithDescription:[NSString stringWithUTF8String:__FUNCTION__]];
                         ^
/Users/jenkins/build/workspace/loopback-sdk-ios/SLRemotingTests/SLRESTAdapterTests.m:67:5: error: use of undeclared identifier 'expectation'
In file included from /Users/jenkins/build/workspace/loopback-sdk-ios/SLRemotingTests/SLRESTAdapterTests.m:13:
/Users/jenkins/build/workspace/loopback-sdk-ios/SLRemotingTests/../SLRemoting/SLRemotingTestsUtils.h:13:45: note: expanded from macro 'ASYNC_TEST_START'
#define ASYNC_TEST_START XCTestExpectation *expectation = [self expectationWithDescription:[NSString stringWithUTF8String:__FUNCTION__]];
                                            ^
/Users/jenkins/build/workspace/loopback-sdk-ios/SLRemotingTests/SLRESTAdapterTests.m:67:5: error: no visible @interface for 'SLRESTAdapterTests' declares the selector 'expectationWithDescription:'
    ASYNC_TEST_START
    ^~~~~~~~~~~~~~~~
In file included from /Users/jenkins/build/workspace/loopback-sdk-ios/SLRemotingTests/SLRESTAdapterTests.m:13:
/Users/jenkins/build/workspace/loopback-sdk-ios/SLRemotingTests/../SLRemoting/SLRemotingTestsUtils.h:13:65: note: expanded from macro 'ASYNC_TEST_START'
#define ASYNC_TEST_START XCTestExpectation *expectation = [self expectationWithDescription:[NSString stringWithUTF8String:__FUNCTION__]];
                                                           ~~~~ ^~~~~~~~~~~~~~~~~~~~~~~~~~
/Users/jenkins/build/workspace/loopback-sdk-ios/SLRemotingTests/SLRESTAdapterTests.m:76:31: error: use of undeclared identifier 'expectation'
                              ASYNC_TEST_SIGNAL
                              ^
In file included from /Users/jenkins/build/workspace/loopback-sdk-ios/SLRemotingTests/SLRESTAdapterTests.m:13:
/Users/jenkins/build/workspace/loopback-sdk-ios/SLRemotingTests/../SLRemoting/SLRemotingTestsUtils.h:23:28: note: expanded from macro 'ASYNC_TEST_SIGNAL'
#define ASYNC_TEST_SIGNAL [expectation fulfill];
                           ^
/Users/jenkins/build/workspace/loopback-sdk-ios/SLRemotingTests/SLRESTAdapterTests.m:78:35: error: use of undeclared identifier 'expectation'
                          failure:ASYNC_TEST_FAILURE_BLOCK];
                                  ^
In file included from /Users/jenkins/build/workspace/loopback-sdk-ios/SLRemotingTests/SLRESTAdapterTests.m:13:
/Users/jenkins/build/workspace/loopback-sdk-ios/SLRemotingTests/../SLRemoting/SLRemotingTestsUtils.h:31:6: note: expanded from macro 'ASYNC_TEST_FAILURE_BLOCK'
    [expectation fulfill]; \
     ^
/Users/jenkins/build/workspace/loopback-sdk-ios/SLRemotingTests/SLRESTAdapterTests.m:79:5: error: no visible @interface for 'SLRESTAdapterTests' declares the selector 'waitForExpectationsWithTimeout:handler:'
    ASYNC_TEST_END
    ^~~~~~~~~~~~~~
In file included from /Users/jenkins/build/workspace/loopback-sdk-ios/SLRemotingTests/SLRESTAdapterTests.m:13:
/Users/jenkins/build/workspace/loopback-sdk-ios/SLRemotingTests/../SLRemoting/SLRemotingTestsUtils.h:18:30: note: expanded from macro 'ASYNC_TEST_END'
#define ASYNC_TEST_END [self waitForExpectationsWithTimeout:10 handler:nil];
                        ~~~~ ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Users/jenkins/build/workspace/loopback-sdk-ios/SLRemotingTests/SLRESTAdapterTests.m:83:5: error: use of undeclared identifier 'XCTestExpectation'
    ASYNC_TEST_START
    ^
In file included from /Users/jenkins/build/workspace/loopback-sdk-ios/SLRemotingTests/SLRESTAdapterTests.m:13:
/Users/jenkins/build/workspace/loopback-sdk-ios/SLRemotingTests/../SLRemoting/SLRemotingTestsUtils.h:13:26: note: expanded from macro 'ASYNC_TEST_START'
#define ASYNC_TEST_START XCTestExpectation *expectation = [self expectationWithDescription:[NSString stringWithUTF8String:__FUNCTION__]];
                         ^
fatal error: too many errors emitted, stopping now [-ferror-limit=]
20 errors generated.
```

hideya commented:
@rmg thanks for the help!  I'm looking into the issue...

hideya commented:
@rmg somehow my recent commits to this PR don't seem to be taken by the test execution on CI...  
With the last commit https://github.com/hideya/loopback-sdk-ios/commit/2e63083a94f4388d529b8af019b075aabc1eef0e , the script should just exit after showing a simple message, but the tests run and fail.  Checking [The console output](http://ci.strongloop.com/job/loopback-sdk-ios/202/console), it seems executing an old version of the script (looks like the one of https://github.com/hideya/loopback-sdk-ios/commit/d6432ecb5d4aa90beb7f2a7444635c8dcba59547)
Would you be able to point out what went wrong...?

rmg commented:
@slnode test please
@hideya sorry, it looks like I accidentally rolled back one of the changes I made to get it working.

hideya commented:
@rmg thank you for your prompt action!  Now it looks working as expected!

hideya commented:
@rmg @bajtos Now the unit tests run on iPhone simulator successfully.  However, there were difficulties to run the tests against OS X on CI, and I'd like to propose skipping it for now (just buildability is confirmed on CI, whereas the tests get executed on a local machine). 

The symptom was the same as https://github.com/bitrise-io/bitrise.io/issues/5
I tried all the workarounds discussed there but was not able to manage running the tests successfully on the CI.  I also recreated another OS X build target using Xcode 5.1.1 (the same version as CI) and tried it, but it didn't work either.  Since OS X support is not official and I'm spending a bit too much time on this, I'd like to propose skipping running the unit tests on OS X on CI for now.  If it is OK, I'll file an Issue about it as a record.

rmg commented:
@hideya that works for me :-)

And thanks for taking this on!

hideya commented:
I'm going to merge this and file a Issue regarding not testing OS X build on the CI.
